test/Test/Database/LSMTree/Internal/Lookup.hs

```diff
@@ -348,48 +446,62 @@ tabulateInMemLookupData dat run =
 
-data InMemLookupData k v = InMemLookupData {
+data InMemLookupData k v b = InMemLookupData {
     -- | Data for constructing a run
-    runData :: Map k (Entry v BlobSpan)
+    runData :: Map k (Entry v b)
     -- | Lookups, with expected return values
   , lookups :: [k]
   }
   deriving stock (Show, Generic)
   deriving anyclass NFData
 
-instance Arbitrary (InMemLookupData SerialisedKey SerialisedValue) where
-  arbitrary = liftArbitrary2InMemLookupData genSerialisedKey genSerialisedValue
-  shrink = liftShrink2InMemLookupData shrinkSerialisedKey shrinkSerialisedValue
+instance Arbitrary (InMemLookupData SerialisedKey SerialisedValue BlobSpan) where
+  arbitrary = liftArbitrary3InMemLookupData genSerialisedKey genSerialisedValue genBlobSpan
+  shrink = liftShrink3InMemLookupData shrinkSerialisedKey shrinkSerialisedValue shrinkBlobSpan
 
-instance Arbitrary1 (InMemLookupData SerialisedKey) where
-  liftArbitrary = liftArbitrary2InMemLookupData genSerialisedKey
+instance Arbitrary1 (InMemLookupData SerialisedKey SerialisedValue) where
+  liftArbitrary = liftArbitrary3InMemLookupData genSerialisedKey genSerialisedValue
 
-liftArbitrary2InMemLookupData :: Ord k => Gen k -> Gen v -> Gen (InMemLookupData k v)
-liftArbitrary2InMemLookupData genKey genValue = do
+instance Arbitrary2 (InMemLookupData SerialisedKey) where
+  liftArbitrary2 = liftArbitrary3InMemLookupData genSerialisedKey
+
+liftArbitrary3InMemLookupData ::
+     Ord k
+  => Gen k
+  -> Gen v
+  -> Gen b
+  -> Gen (InMemLookupData k v b)
+liftArbitrary3InMemLookupData genKey genValue genBlob = do
     kops <- liftArbitrary2Map genKey (liftArbitrary genEntry)
               `suchThat` (\x -> Map.size (Map.filter isJust x) > 0)
     let runData = Map.mapMaybe id kops
-    lookups <- sublistOf (Map.keys kops) >>= shuffle
+    lookups <- (sublistOf (Map.keys kops) >>= shuffle)
     pure InMemLookupData{ runData, lookups }
   where
-    genEntry = liftArbitrary2 genValue arbitrary
-
-liftShrink2InMemLookupData :: Ord k => (k -> [k]) -> (v -> [v]) -> InMemLookupData k v -> [InMemLookupData k v]
-liftShrink2InMemLookupData shrinkKey shrinkValue InMemLookupData{ runData, lookups } =
+    genEntry = liftArbitrary2 genValue genBlob
+
+liftShrink3InMemLookupData ::
+     Ord k
+  => (k -> [k])
+  -> (v -> [v])
+  -> (b -> [b])
+  -> InMemLookupData k v b
+  -> [InMemLookupData k v b]
+liftShrink3InMemLookupData shrinkKey shrinkValue shrinkBlob InMemLookupData{ runData, lookups } =
          [ InMemLookupData runData' lookups
          | runData' <- liftShrink2Map shrinkKey shrinkEntry runData
          , Map.size runData' > 0 ]
       ++ [ InMemLookupData runData lookups'
          | lookups' <- liftShrink shrinkKey lookups ]
     where
-      shrinkEntry = liftShrink2 shrinkValue shrink
+      shrinkEntry = liftShrink2 shrinkValue shrinkBlob
 
 genSerialisedKey :: Gen SerialisedKey
 genSerialisedKey = arbitrary `suchThat` (\k -> sizeofKey k >= 6)
```

Comment:
At some point we can think about unifying approaches for generation of these core types, but might make sense to wait until we see what @dcoutts has come up with his `LogicalPage` related generators.

Reply:
Let's wait for @dcoutts